### PR TITLE
Operator functions.

### PR DESCRIPTION
**Pull request by izulin**
### Context

New functions: HF.ADD, HF.CONCAT, HF.DIVIDE, HF.EQ, HF.GT, HF.GTE, HF.LT, HF.LTE, HF.MINUS, HF.MULTIPLY, HF.NE, HF.POW, HF.UMINUS, HF.UNARY_PERCENT, HF.UPLUS.
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: test/interpreter/function-concat.spec.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      ['=CONCAT(1)', '=CONCAT(1, 1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=CONCAT("a","b")'],
```
Comment (bardek8) [inline test/interpreter/function-concat.spec.ts:17]: Maybe we can add a test with string longer than 1 character :)

File: test/interpreter/function-divide.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=DIVIDE(6,3)'],
+      ['=DIVIDE(,1)'],
+      ['=DIVIDE(,)']
```
Comment (bardek8) [inline test/interpreter/function-divide.spec.ts:19]: what is the default if the second argument is missing?

File: test/interpreter/function-divide.spec.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      ['=DIVIDE(1)', '=DIVIDE(1, 1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=DIVIDE(6,3)'],
```
Comment (bardek8) [inline test/interpreter/function-divide.spec.ts:17]: what about non-integer division?

File: test/interpreter/function-uplus.spec.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      ['=UPLUS()', '=UPLUS(1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=UPLUS(2)'],
```
Comment (bardek8) [inline test/interpreter/function-uplus.spec.ts:17]: UPLUS(-3)?

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+export class SimpleArithmerticPlugin extends  FunctionPlugin {
+  public static implementedFunctions = {
+    'ADD': {
+      method: 'add',
+      parameters: [
+        { argumentType: ArgumentTypes.NUMBER },
+        { argumentType: ArgumentTypes.NUMBER },
+      ],
+    },
+    'CONCAT' : {
```
Comment (bardek8) [inline src/interpreter/plugin/SimpleArithmertic.ts:20]: We already have CONCATENATE in text functions. Maybe we can have this as an alias?

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+      ]
+    },
+    'GT': {
+      method: 'gt',
+      parameters: [
+        { argumentType: ArgumentTypes.NOERROR },
+        { argumentType: ArgumentTypes.NOERROR },
+      ]
+    },
+    'GEQ': {
```
Comment (bardek8) [inline src/interpreter/plugin/SimpleArithmertic.ts:48]: sed s/GEQ/GTE/g

File: test/interpreter/function-uminus.spec.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      ['=UMINUS()', '=UMINUS(1, 1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('B1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should calculate the correct value with correct defaults', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=UMINUS(2)'],
```
Comment (bardek8) [inline test/interpreter/function-uminus.spec.ts:17]: uminus(-3)

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+      ]
+    },
+    'LT': {
+      method: 'lt',
+      parameters: [
+        { argumentType: ArgumentTypes.NOERROR },
+        { argumentType: ArgumentTypes.NOERROR },
+      ]
+    },
+    'LEQ': {
```
Comment (bardek8) [inline src/interpreter/plugin/SimpleArithmertic.ts:62]: sed s/LEQ/LTE/g

**Comment by izulin** [comment]
there is an issue: in Czech, we have NOT: 'NE', and NE: 'NE'

**Review by bardek8 — CHANGES_REQUESTED** [review]
Apart from the in-line comments, we are missing translations. Please add //FIXME everywhere.

File: test/interpreter/function-divide.spec.ts (hunk 2 above)
Comment (izulin) [reply test/interpreter/function-divide.spec.ts:19]: it's not even default, it's standard null->number coercion, so 0

File: src/interpreter/plugin/SimpleArithmertic.ts (hunk 5 above)
Comment (izulin) [reply src/interpreter/plugin/SimpleArithmertic.ts:20]: CONCATENATE is to CONCAT as SUM is to ADD
(many args vs 2 args)

File: test/interpreter/function-uminus.spec.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      ['=UMINUS(-3)'],
```
Comment (wojciechczerniak) [inline test/interpreter/function-uminus.spec.ts:18]:
```suggestion
      ['=UMINUS(-3)'],
      ['=UMINUS(0)']
```

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
All of those functions are `formula.js` specific. They don't exist in Excel (except CONCAT but it's a different concat) or ODFF. We will not have translations for them. Therefore we've discussed with @krzysztofspilka to prefix all those [Nonstandard Function Names](https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1017944_715980110) with our vendor prefix `HF.` and avoid naming conflicts.

Those are nonstandard functons that are based on JavaScript types and logic.

File: test/interpreter/function-concat.spec.ts [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  it('should throw correct error', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=CONCAT(NA(),)'],
+      ['=CONCAT(B2:C2,)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA))
+    expect(engine.getCellValue(adr('A2'))).toEqual(detailedError(ErrorType.VALUE, ErrorMessage.WrongType))
+  })
+})
```
[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply test/interpreter/function-concat.spec.ts:46]: Nevermind! This is not the same CONCAT that new Excel provides

File: src/interpreter/plugin/SimpleArithmertic.ts (hunk 5 above)
Comment (wojciechczerniak) [reply src/interpreter/plugin/SimpleArithmertic.ts:20]: Yup, this is not the Excel CONCAT and CONCATENATE

File: test/interpreter/function-concat.spec.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual('TRUE')
+    expect(engine.getCellValue(adr('A2'))).toEqual('1')
+  })
+
+  it('should throw correct error', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=CONCAT(NA(),)'],
+      ['=CONCAT(B2:C2,)'],
```
[the remark this answers is not in the document]
Comment (izulin) [reply test/interpreter/function-concat.spec.ts:40]: it is not for the & operator

File: test/interpreter/function-uminus.spec.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+      ['=UMINUS(0)']
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(-2)
+    expect(engine.getCellValue(adr('A2'))).toEqual(3)
```
Comment (izulin) [inline test/interpreter/function-uminus.spec.ts:23]:
```suggestion
    expect(engine.getCellValue(adr('A2'))).toEqual(3)
    expect(engine.getCellValue(adr('A3'))).toEqual(0)
```

**Comment by izulin** [comment]
> All of those functions are `formula.js` specific. They don't exist in Excel (except CONCAT but it's a different concat) or ODFF. We will not have translations for them. Therefore we've discussed with @krzysztofspilka to prefix all those [Nonstandard Function Names](https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1017944_715980110) with our vendor prefix `HF.` and avoid naming conflicts.
> 
> Those are nonstandard functons that are based on JavaScript types and logic.

HF.ADD and HF.UNARY_PERCENT ?

File: test/interpreter/function-concat.spec.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply test/interpreter/function-concat.spec.ts:40]: True. I forgot to cross this one out. This is not the https://support.microsoft.com/en-us/office/concat-function-9b1a9a3f-94ff-41af-9736-694cbd6b4ca2 function. It's a `formula.js` alias for `&` operator

**Comment by wojciechczerniak** [comment]
> > All of those functions are `formula.js` specific. They don't exist in Excel (except CONCAT but it's a different concat) or ODFF. We will not have translations for them. Therefore we've discussed with @krzysztofspilka to prefix all those [Nonstandard Function Names](https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1017944_715980110) with our vendor prefix `HF.` and avoid naming conflicts.
> > Those are nonstandard functons that are based on JavaScript types and logic.
> 
> HF.ADD and HF.UNARY_PERCENT ?

YES

File: src/interpreter/plugin/SimpleArithmertic.ts (hunk 5 above)
Comment (bardek8) [reply src/interpreter/plugin/SimpleArithmertic.ts:20]: Good point, sorry!

**Comment by bardek8** [comment]
As for adding `HF`, let me mention that the functions do appear in GoogleSheets (https://support.google.com/docs/answer/3093590?hl=en&ref_topic=3105518). Probably you've already noticed that, but I wanted to make sure :)

File: test/interpreter/function-divide.spec.ts (hunk 2 above)
Comment (bardek8) [reply test/interpreter/function-divide.spec.ts:19]: Sure! I just used the "defaults" from the test name :) Can we have a tiny test for that: `DIVIDE(4,) -> DIV0` ?

**Review by bardek8 — APPROVED** [review]

File: docs/guide/built-in-functions.md [hunk 13]
```diff
@@ -189,6 +189,21 @@ lets you design your own [custom functions](custom-functions).
...
+| HF.LT | Math and trigonometry | Tests two values for less-than relation. | HF.LT(Value; Value) |
+| HF.GTE | Math and trigonometry | Tests two values for greater-equal relation. | HF.GEQ(Value; Value) |
+| HF.GT | Math and trigonometry | Tests two values for greater-than relation. | HF.GT(Value; Value) |
+| HF.MINUS | Math and trigonometry | Subtracts two values. | HF.MINUS(Number; Number) |
+| HF.MULTIPLY | Math and trigonometry | Multiplies two values. | HF.MULTIPLY(Number; Number) |
+| HF.NE | Math and trigonometry | Tests two values for inequality. | HF.NE(Value; Value) |
+| HF.POW | Math and trigonometry | Computes power of two values. | HF.POW(Number; Number) |
+| HF.UMINUS | Math and trigonometry | Negates the value. | HF.UMINUS(Number) |
+| HF.UNARY_PERCENT | Math and trigonometry | Applies percent operator. | HF.UNARY_PERCENT(Number) |
+| HF.UPLUS | Math and trigonometry | Applies unary plus. | HF.UPLUS(Number) |
```
Comment (wojciechczerniak) [inline docs/guide/built-in-functions.md:206]:
```suggestion
| HF.ADD | Operator | Adds two values. | HF.ADD(Number; Number) |
| HF.CONCAT | Operator | Concatenates two strings. | HF.CONCAT(String; String) |
| HF.DIVIDE | Operator | Divides two values. | HF.DIVIDE(Number; Number) |
| HF.EQ | Operator | Tests two values for equality. | HF.EQ(Value; Value) |
| HF.LTE | Operator | Tests two values for less-equal relation. | HF.LEQ(Value; Value) |
| HF.LT | Operator | Tests two values for less-than relation. | HF.LT(Value; Value) |
| HF.GTE | Operator | Tests two values for greater-equal relation. | HF.GEQ(Value; Value) |
| HF.GT | Operator | Tests two values for greater-than relation. | HF.GT(Value; Value) |
| HF.MINUS | Operator | Subtracts two values. | HF.MINUS(Number; Number) |
| HF.MULTIPLY | Operator | Multiplies two values. | HF.MULTIPLY(Number; Number) |
| HF.NE | Operator | Tests two values for inequality. | HF.NE(Value; Value) |
| HF.POW | Operator | Computes power of two values. | HF.POW(Number; Number) |
| HF.UMINUS | Operator | Negates the value. | HF.UMINUS(Number) |
| HF.UNARY_PERCENT | Operator | Applies percent operator. | HF.UNARY_PERCENT(Number) |
| HF.UPLUS | Operator | Applies unary plus. | HF.UPLUS(Number) |
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added 9 text functions EXACT, LOWER, UPPER, MID, T, SUBSTITUTE, REPLACE, UNICODE, UNICHAR. (#159)
 - Added 5 datetime functions: INTERVAL, NETWORKDAYS, NETWORKDAYS.INTL, WORKDAY, WORKDAY.INTL. (#153)
 - Added 3 information functions HLOOKUP, ROW, COLUMN. (#158)
+- Added 15 math functions HF.ADD, HF.CONCAT, HF.DIVIDE, HF.EQ, HF.GT, HF.GEQ, HF.LT, HF.LEQ, HF.MINUS, HF.MULTIPLY, HF.NE, HF.POW, HF.UMINUS, HF.UNARY_PERCENT, HF.UPLUS.
```
Comment (wojciechczerniak) [inline CHANGELOG.md:13]:
```suggestion
- Added 15 operator functions HF.ADD, HF.CONCAT, HF.DIVIDE, HF.EQ, HF.GT, HF.GEQ, HF.LT, HF.LEQ, HF.MINUS, HF.MULTIPLY, HF.NE, HF.POW, HF.UMINUS, HF.UNARY_PERCENT, HF.UPLUS.
```

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+
+  public minus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.MINUS'),
+      (a: number, b: number) => this.interpreter.arithmeticHelper.subtract(a, b)
+    )
+  }
+
+  public multiply(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.MULTIPLY'),
+      (a: number, b: number) => a*b
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:183]: Can we refactor this fragment and TIMES_OP https://github.com/handsontable/hyperformula/blob/master/src/interpreter/Interpreter.ts#L146 to a common `this.interpreter.arithmeticHelper.multiply(a, b)` helper?

Something we do for `(a: number, b: number) => this.interpreter.arithmeticHelper.subtract(a, b)`. 

The reasoning is to have a single place for computation, both are basicaly the same. Linking them with a shared helper would make developers aware of that in the future.

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+  //eslint-disable-next-line @typescript-eslint/camelcase
+  public unary_percent(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UNARY_PERCENT'),
+      (a => a/100)
+    )
+  }
+
+  public uplus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UPLUS'),
+      (a => a)
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:214]: Same

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+  public uminus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UMINUS'),
+      (a => -a)
+    )
+  }
+
+  //eslint-disable-next-line @typescript-eslint/camelcase
+  public unary_percent(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UNARY_PERCENT'),
+      (a => a/100)
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:208]: Same

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+    )
+  }
+
+  public pow(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.POW'), Math.pow)
+  }
+
+  public uminus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UMINUS'),
+      (a => -a)
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:201]: Same

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 19]
```diff
@@ -0,0 +1,217 @@
...
+  public ne(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.NE'),
+      (a: InternalNoErrorCellValue, b: InternalNoErrorCellValue) => (
+        this.interpreter.arithmeticHelper.compare(a, b) !== 0
+      )
+    )
+  }
+
+  public pow(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.POW'), Math.pow)
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:196]: Same

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+      (a: InternalNoErrorCellValue, b: InternalNoErrorCellValue) => (
+        this.interpreter.arithmeticHelper.compare(a, b) <= 0
+      )
+    )
+  }
+
+  public minus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.MINUS'),
+      (a: number, b: number) => this.interpreter.arithmeticHelper.subtract(a, b)
+    )
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:178]: Would passing a reference to the helper method be simple and enough?
```suggestion
    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.MINUS'), this.interpreter.arithmeticHelper.subtract)
```

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+
+  public add(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.ADD'),
+      (a: number, b: number) => this.interpreter.arithmeticHelper.addWithEpsilon(a, b)
+      )
+  }
+
+  public concat(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.CONCAT'),
+      (a: string, b: string) => a.concat(b)
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:125]: Same. I would connect it somehow with CONCATENATE_OP

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 22]
```diff
@@ -0,0 +1,217 @@
...
+      parameters: [
+        { argumentType: ArgumentTypes.NUMBER },
+      ],
+    },
+  }
+
+  public add(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.ADD'),
+      (a: number, b: number) => this.interpreter.arithmeticHelper.addWithEpsilon(a, b)
+      )
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:120]: If passing ref will work, we can do the same here
```suggestion
    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.ADD'), this.interpreter.arithmeticHelper.addWithEpsilon)
```

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 23]
```diff
@@ -0,0 +1,217 @@
...
+
+  public concat(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.CONCAT'),
+      (a: string, b: string) => a.concat(b)
+    )
+  }
+
+  public divide(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.DIVIDE'),
+      divide
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:131]: Why `divide` is exported differently from the arithmetic helper? We can `this.interpreter.arithmeticHelper.subtract` but not `this.interpreter.arithmeticHelper.divide` 🙄 We could import all the functions from arithmetic helper and would be correct as well, but can we use one, consistent manner?

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]
> As for adding `HF`, let me mention that the functions do appear in GoogleSheets (https://support.google.com/docs/answer/3093590?hl=en&ref_topic=3105518). Probably you've already noticed that, but I wanted to make sure :)

Right! But GS will create a problem with CONCAT function anyway. We can use the GS approach to categorize this functions as Operator, thanks for the idea 👍 

And it will be easy to be GS compilant with custom language file, just to rename / alias the functions.

File: src/interpreter/plugin/SimpleArithmertic.ts (hunk 23 above)
Comment (izulin) [reply src/interpreter/plugin/SimpleArithmertic.ts:131]: Only functions that need context (e.g. config) are part of 'object' in ArithmeticHelper, while context-free are only part of module.

File: src/interpreter/plugin/SimpleArithmertic.ts [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+  }
+
+  public uminus(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('HF.UMINUS'),
+      this.interpreter.arithmeticHelper.unaryMinus
+    )
+  }
+
+  //eslint-disable-next-line @typescript-eslint/camelcase
+  public unary_percent(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
```
Comment (wojciechczerniak) [inline src/interpreter/plugin/SimpleArithmertic.ts:195]: Does it have to be an exception in the naming convention? I think that the compatiblity with GS and FORMULA.JS is handled at the metadata key / language level?

**Review by wojciechczerniak — APPROVED** [review]
This look so clean now. Thanks! ❤️

**Pull request merged**